host/terminal_conin.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2016, Intel Corporation
+ * All rights reserved.
+ *
+ * Author: Léo Sartre <leo.sartre@intel.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer
+ *      in the documentation and/or other materials provided with the
+ *      distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <termios.h>
+#include <unistd.h>
+#include <ewlog.h>
+
+#include "terminal_conin.h"
+#include "worker.h"
+
+static EFIAPI VOID
+terminal_conin_wait_for_key(__attribute__((__unused__)) EFI_EVENT Event,
+			    VOID *Context)
+{
+	int n;
+	char buf[3];
+	EFI_SYSTEM_TABLE *st = (EFI_SYSTEM_TABLE *)Context;
+
+	n = read(STDIN_FILENO, buf, sizeof(buf)*sizeof(buf[0]));
```

Comment:
It is to catch properly consume ANSI escape sequences \033[A ... \033[D emited when you press arrow keys.
Maybe I can make a comment here or a #define ANSI_ESCAPE_LEN 3, which one do you prefer?